messaging/amqp/src/main/java/io/quarkus/ts/openshift/messaging/amqp/PriceConsumer.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.ts.openshift.messaging.amqp;
+
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.jboss.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@ApplicationScoped
+public class PriceConsumer {
+
+    private static final Logger LOG = Logger.getLogger(PriceProducer.class.getName());
+
+    private final List<Integer> prices = Collections.synchronizedList(new ArrayList<>());
+
+    @Incoming("prices")
+    public void process(Integer price) {
+        LOG.info("process fired: " + price);
+        this.prices.add(price);
+    }
+
+    public List<Integer> getPrices() {
+        return prices;
```

Comment:
Do you know what bug happens here?

Reply:
@Ladicek Hmm...it probably should give away the reference to the original list and it should create a new one to return? Not sure.

Reply:
`prices` is `Collections.synchronizedList`, so iteration over it must also be synchronized. You can either return a defensive copy here, or get rid of `synchronizedList` and use something like `ConcurrentLinkedQueue` or `CopyOnWriteArrayList`.

Reply:
@Ladicek Thanks. The latest commit addresses that.